learn/get-started/write-a-graphql-api-with-ballerina.md

```diff
@@ -14,49 +14,72 @@ redirect_from:
-`Query` type will be used to read the data in the data source, and the `Mutation` operation will be used to update the
-data in the data source.
+The GraphQL endpoint you will create in this guide will have two main operations, `query` and `mutation`. The `query` type will be used to read the data in the data source, and the `mutation` operation will be used to update the data in the data source.
 
-### The `Query` Type
-The `Query` type has two fields.
+### The `query` type
+The `query` type has two fields.
+* Can be identified by the `resource` keyword in the function description.
 * The `all` field - This field will return all the data in the data source as an array.
-
 * The `filter` field - This field will return the data filtered by the ISO Code of a country(`isoCode`).
 
-### The `Mutation` Type
-The `Mutation` type has a single field.
+### The `mutation` type
+The `mutation` type has a single field.
+* Can be identified by the `remote` keyword in the function description.
 * The `add` field - This field will add a given entry to the data source.
 
-## Create a data source for the project
+## Create the service package
+
+Ballerina uses packages to group code. You need to create a Ballerina package, generate the service code in the package, and write the business logic.
+
+>**Info:** For more information on Ballerina packages, see [Organize Ballerina code](/learn/organize-ballerina-code).
+
+In the terminal, execute the command below to create the Ballerina package for the API implementation.
+
+```bash
+$ bal new graphql
+```
 
-Before writing the GraphQL service, let's create a data source for the project. This will mimic a database that stores
-the data for the service. In this guide, you are going to use an in-memory table in Ballerina as the data source.
+You view the output below.
+
+```bash
+Created new package 'graphql' at graphql.
+```
+
+This creates a directory named `graphql` with the default module along with a sample code for the service as shown below.
+
+```bash
+.
+├── graphql
+│   ├── Ballerina.toml
+│   └── main.bal
+```
+
+>**Tip:** Remove the automatically-created `main.bal` file as you are not going to use it in this guide.
+
+## Create the data source 
+
+Before writing the GraphQL service, let's create a data source for the project. This will mimic a database that stores the data for the service using an in-memory table in Ballerina as the data source.
 
 ### Define the types for the data source
 
- To create the table, you have to define the data type to use in the table. The following code defines the record type
- `CovidEntry`. A single entry will include the ISO code for the location (`isoCode`), name of the country (`country`),
- number of cases (`cases`), number of deaths (`deaths`), number of recovered cases (`recovered`), and the number of
- active cases (`active`).
+To define the data type to use in the table, replace the API template file (i.e., `main.bal`) with the code below. 
```

Comment:
in the previous tip it was asekd to remove the `main.bal`. Also if we use `-t service` as I suggested in a previous comment there won't be `main.bal` file.

Reply:
Updated all the occurrences of the `main.bal` file to `service.bal`.